src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -190,7 +188,7 @@ private void CommitNode(long blockNumber, Hash256? address, ref TreePath path, i
 
                 if (!_pruningStrategy.PruningEnabled)
                 {
-                    PersistNode(address, path, node, blockNumber, writeFlags);
+                    PersistNode(address, path, node, blockNumber, writeBatch!, writeFlags);
```

Comment:
writeBatch can be null, but we are assuming here it is now null, probably because of `NonPruningTrieStoreCommitter`, but can we express this better? Maybe move this method to the Committer?

Reply:
Yea, I originally split it out, but then merge it back in to reduce change.

Reply:
Move it back out?

Reply:
Done